code/game/machinery/vending/_vending.dm

```diff
@@ -124,22 +124,9 @@
 	
 
 /obj/machinery/vending/ex_act(severity)
-	switch(severity)
-		if(1.0)
-			qdel(src)
-			return
-		if(2.0)
-			if (prob(50))
-				qdel(src)
-				return
-		if(3.0)
-			if (prob(25))
-				spawn(0)
-					src.malfunction()
-					return
-				return
-		else
-	return
+	if(severity == 3 && prob(25))
+		malfunction()
```

Comment:
also here

Reply:
Ditto, it does malfunction check then runs parent which should handle the wholesale destruction.